### DISCUSSION TITLE
Kotlin Multiplatform implementations

### DISCUSSION THREAD

**Discussion by gravelld** [opensubsonic/open-subsonic-api#97] [General]
(Hope it's ok to add this rather general question that isn't specifically about the _development_ of the API)...

Anyone know any client implementations of OpenSubsonic and/or (legacy) Subsonic in Kotlin Multiplatform? I guess simple SDKs wrapped around the HTTP APIs.

Anyone interested in collaborating on one?

**Comment by MrNuggelz** [comment]
I started working on a Kotlin Multiplatform client implementation. It's still a work in progress, but my plan is to first implement all API endpoints that are provided by navidrome, as currenty not all API endpoints have an example response/structure provided in the OpenSubsonic documentation. You can see my current progress here: https://github.com/MrNuggelz/OpenSubsonicClient

**Reply by gravelld** [reply]
Looks good! It also looks like you have done quite a lot of the original Subsonic spec...
